### PR TITLE
Add template for proxy plugin

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
- [x] Add proxy plugin as a part of openvino
- [x] Add tests with mock plugins
- [x] Implement import model
- [x] Support query model
- [x] Support add extension
- [x] Cover load with split or without split
- [x] Change plugins.xml structure
- [x] Try to run on CPU
- [ ] Try to run on GPU

### Tickets:
 - 78226

### CODE REVIEW

File: src/plugins/proxy/src/plugin.cpp [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+#include "cpp_interfaces/interface/ie_iplugin_internal.hpp"
+#include "ie_icore.hpp"
+#include "openvino/runtime/common.hpp"
+#include "proxy_plugin.hpp"
+
+ov::proxy::Plugin::Plugin() {}
+ov::proxy::Plugin::~Plugin() {}
+
+void ov::proxy::Plugin::SetConfig(const std::map<std::string, std::string>& config) {
+    m_config = config;
```
Comment (ilyachur) [inline src/plugins/proxy/src/plugin.cpp:19]: Add to primary plugin

File: src/plugins/proxy/src/plugin.cpp [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+}
+InferenceEngine::QueryNetworkResult ov::proxy::Plugin::QueryNetwork(
+    const InferenceEngine::CNNNetwork& network,
+    const std::map<std::string, std::string>& config) const {
+    IE_THROW(NotImplemented);
```
Comment (ilyachur) [inline src/plugins/proxy/src/plugin.cpp:24]: Need to implement for case HETERO:MOCK,CPU

? Just recall internal plugins

**Comment by lohika-denis-kotov** [comment]
May I ask, what is the purpose of Proxy plugin ?

**Comment by ilya-lavrenov** [comment]
> @ilyachur May I ask, what is a purpose of Proxy plugin ?

- Run NVIDIA and Intel GPUs under a common GPU name. E.g. machine with single Intel & singe NVIDIA GPUs will see the devices like GPU.0 (Intel) and GPU.1 (NVIDIA)
- NVIDIA GPU will fallback to OpenCL implementations from Intel GPU plugin, which allow to saturate NVIDIA device more. It will be done via HETERO plugin automatically

**Comment by ilyachur** [comment]
> @ilyachur May I ask, what is a purpose of Proxy plugin ?

@lohika-denis-kotov, @ilya-lavrenov answered. You can also find additional information on the page: https://github.com/openvinotoolkit/openvino/wiki/Proxy-plugin-%28concept%29

**Comment by lohika-denis-kotov** [comment]
> > @ilyachur May I ask, what is a purpose of Proxy plugin ?
> 
>     * Run NVIDIA and Intel GPUs under a common GPU name. E.g. machine with single Intel & singe NVIDIA GPUs will see the devices like GPU.0 (Intel) and GPU.1 (NVIDIA)
> 
>     * NVIDIA GPU will fallback to OpenCL implementations from Intel GPU plugin, which allow to saturate NVIDIA device more. It will be done via HETERO plugin automatically

@ilya-lavrenov Thanks !!

**Comment by akladiev** [comment]
This PR was closed because it has been stalled for 2 week with no activity.

**Pull request closed without merging**